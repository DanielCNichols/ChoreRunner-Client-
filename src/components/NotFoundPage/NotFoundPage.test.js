import React from 'react';
import ReactDOM from 'react-dom';
import NotFoundPage from './NotFoundPage';
import {MemoryRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'

describe.only('NotFoundPage component testing', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
  
    ReactDOM.render(<MemoryRouter><NotFoundPage /></MemoryRouter>, div);
  
    ReactDOM.unmountComponentAtNode(div);
  });
  
  it('renders UI as expected', () => {
    const tree = renderer
        .create(<MemoryRouter><NotFoundPage /></MemoryRouter>)
        .toJSON();
        expect(tree).toMatchSnapshot();
  })

  it('renders empty given no tabs', () => {
    const wrapper = shallow(<NotFoundPage />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
