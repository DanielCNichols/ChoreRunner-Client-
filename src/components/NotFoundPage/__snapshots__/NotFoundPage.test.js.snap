// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage component testing renders UI as expected 1`] = `
<div
  className="notfound-container"
>
  <h2>
    Page not found
  </h2>
</div>
`;

exports[`NotFoundPage component testing renders empty given no tabs 1`] = `
<div
  className="notfound-container"
>
  <h2>
    Page not found
  </h2>
</div>
`;
